src/main/java/com/svydovets/bibirnate/cache/Cache.java

```diff
@@ -0,0 +1,164 @@
+package com.svydovets.bibirnate.cache;
+
+
+import com.svydovets.bibirnate.cache.command.KeyExtractorCommand;
+import com.svydovets.bibirnate.cache.command.impl.EntityKeyExtractorCommand;
+import com.svydovets.bibirnate.cache.command.impl.QueryKeyExtractorCommand;
+import com.svydovets.bibirnate.cache.command.invalidation.InvalidationCommand;
+import com.svydovets.bibirnate.cache.command.invalidation.impl.EntityKeyInvalidationCommand;
+import com.svydovets.bibirnate.cache.command.invalidation.impl.QueryKeyInvalidationCommand;
+import com.svydovets.bibirnate.cache.key.Key;
+import com.svydovets.bibirnate.cache.key.parameters.AbstractKeyParam;
+import com.svydovets.bibirnate.exception.CacheOverloadException;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ *
+ */
+public class Cache {
+
+    private static final int DEFAULT_CACHE_SIZE = 10;
+    private static final int MIN_CACHE_SIZE = 1;
+    private static final int MAX_CACHE_SIZE = 50_000;
+    private static final int ZERO_DAYS = 0;
+    private static final int ONE_DAY = 1;
+    private static final int ONE_HOUR = 1;
+    private static final int TWO_HOURS = 2;
+    private static final int ONE_HUNDRED = 100;
+    private static final int MAX_CACHE_SIZE_PERCENTAGE = 90;
+    private static int CACHE_SIZE;
+    private static final String PARAMETER_CANNOT_BE_NULL = "Parameter [%s] cannot be null!";
+
```

Comment:
Maybe we should have a separate file with constants? For example, PARAMETER_CANNOT_BE_NULL can be in /constants/Validation.java